app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,99 @@
+class FriendshipsController < ApplicationController
+  include LolConnections
+  before_action :search_summoner_for_request_action, only: [:edit]
+  before_action :require_user
+  before_action :require_same_user, only: [:edit, :decide]
+  
+ 
+	def new
+    @user = current_user
```

Comment:
I actually think user should be an instance variable.  The other changes are fine though.  If you make user a local variable then there is no benefit of doing `user ||= current_user` because `user` will have to redefined everytime a new action is carried out.  In reality, the same user will be  logged in regardless of the controller action executing and that variable is used in almost every one, so you would still have to do a db call everytime you want to get that data.   `@user ||= current_user` is the same as saying 
if @user == nil
  @user = current_user
We only want to do that db operation if we have not already defined the user.
Let me know if this doesn't make sense.